Jenkinsfile

```diff
@@ -455,27 +462,33 @@ def promoteServiceSetup(openShiftHost, openShiftToken, svcName,registry,imageNam
         echo "skip dc/svc/route cleanup related exception, the resource may not exist. " + e.getMessage();
     }
     try {
-        sh """
-            oc create dc ${svcName} --image=${registry}/${imageNameSpace}/${svcName}:${tagName} -n ${projName} 2> /dev/null
-            oc set env dc ${svcName} APP_NAME=${svcName} -n ${projName} 2> /dev/null
-            oc rollout cancel dc ${svcName} -n ${projName} 2> /dev/null
-            oc expose dc ${svcName} --port=8080 -n ${projName} 2> /dev/null
-            oc expose svc ${svcName} --name=${svcName} -n ${projName} 2> /dev/null
+        sh """ 
+            oc create dc ${svcName} --image=${registry}/${imageNameSpace}/${svcName}:${tagName} -n ${projName} 2> /dev/null     
+            oc rollout pause dc ${svcName} -n ${projName} 2> /dev/null 
+            oc patch dc ${svcName} -p '{"spec": {"template": {"spec": {"containers": [{"name": "default-container","imagePullPolicy": "Always"}]}}}}' -n ${projName} 2> /dev/null
+            oc set env dc ${svcName} APP_NAME=${svcName} -n ${projName} 2> /dev/null 
+            oc rollout resume dc ${svcName} -n ${projName} 2> /dev/null 
+            oc expose dc ${svcName} --type=ClusterIP  --port=80 --protocol=TCP --target-port=8080 -n ${projName} 2> /dev/null
```

Comment:
this will configure all application at port 80 but the nodejsalert-ui application port should be "8080" same as Dev environment .

Reply:
Service port is 80, and target port is 8080. this is workable  configuration.
Please double check your configuration. If they are wrong, please provide suggested ports.

Reply:
it's fine, no change required. I thought initially that smoke test will fail for node.js app but it's passing successfully on port 8080  "http://nodejsalert-ui:8080".